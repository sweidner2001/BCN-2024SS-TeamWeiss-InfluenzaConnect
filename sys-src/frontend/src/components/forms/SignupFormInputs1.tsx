// React Imports:
import React, {useState} from 'react';
import * as yup from "yup";

// Imports eigene Componenten:
import InputField from "../input/InputField";
import InputMultiSelectDropdown from "../input/InputMultiSelectDropdown";

//___________________ Datentypen  ________________
interface ISignupForm1 {
    form1: any;
}


interface IFormInputs1 {
    email: string;
    passwort: string;
    sprache: string[];
}



//___________________ Formular Validation  ________________
/**
 * @function getMaxFieldLength Gibt die max. Länge zurück und einen Fehlertext für die Formulardaten-Validation
 * @author Sebastian Weidner
 * @since 13.06.2024
 * @version 1.0
 *
 * @param maxLength Maximale Textlänge des input-Feldes
 */
const getMaxFieldLength = (maxLength: number): [number, string] => {
    const maxFieldLengthText = `Eingabefeld ist auf ${maxLength} Zeichen begrenzt`;
    return [maxLength, maxFieldLengthText];
};



const SignupSchema1 = yup.object({
    email: yup.string().trim().email("Bitte geben Sie eine gültige Email-Adresse ein!").required('Bitte geben Sie eine Email an!').max(...getMaxFieldLength(40)),
    passwort: yup.string().required("Bitte geben Sie ein Passwort an!").min(10, "Das Passwort muss min. 10 Zeichen lang sein!").max(...getMaxFieldLength(25))
});


const options: string[] = [
    'Deutsch',
    'Englisch',
    'Französisch',
    'Italienisch',
    'Spanisch',
];


/**
 * @function SignupFormInputs1 Formular-Input-Felder für die Anmeldeinformationen
 * @author Sebastian Weidner
 * @since 14.06.2024
 * @version 1.0
 *
 * @param ISignupForm1 "react-hook-form" Daten
 */
const SignupFormInputs1: React.FC<ISignupForm1> = ({form1}) => {

    //___________________ HTML: Formular ________________
    return (
        <>
            {/*--- Überschrift ---*/}
            <div className="border-b-2 border-gray-900/10 pb-4">
                <h3 className='text-xl font-bold mt-6 mb-0 text-gray-900 '>Anmeldeinformationen</h3>
                <p className="mt-1 pt-0 text-sm leading-6 text-gray-500">
                    Die Anmeldeinformationen sind öffentlich nicht sichtbar!
                </p>
            </div>
            <div className="mt-6 mb-2 grid grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6">

                <InputField id="email" label="Email" type="text"
                            autoComplete="email" fieldWidth={4}
                            register={form1.register("email")} error={form1.formState.errors.email?.message}/>

                <InputField id="passwort" label="Passwort" type="password"
                            autoComplete="current-password" fieldWidth={4}
                            register={form1.register("passwort")} error={form1.formState.errors.passwort?.message}/>


                <InputMultiSelectDropdown options={["a", "b", "c"]}
                                          control={form1.control}
                                          name="controlled"/>
            </div>
        </>
    );
};


//__________ Exporte: ___________
export {
    SignupFormInputs1,
    SignupSchema1
}

export type {
    IFormInputs1
}
