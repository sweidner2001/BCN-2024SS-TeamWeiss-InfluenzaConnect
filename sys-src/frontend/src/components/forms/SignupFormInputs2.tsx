// React Imports:
import React, {useState} from 'react';
import * as yup from "yup";

// Imports eigene Componenten:
import InputField from "../input/InputField";
import InputSelect from "../input/InputSelect";
import InputTextarea from "../input/InputTextarea";
import InputMultiSelectDropdown from "../input/InputMultiSelectDropdown";


//___________________ Datentypen  ________________
interface ISignupForm2 {
    form2: any;
}


interface IFormInputs2 {
    anrede: string;
    vorname: string;
    nachname: string;
    land: string;
    bundesland: string;
    telefonnr: string;
    sprache: string[];
    ueberMich: string;
}


//___________________ Formular Validation  ________________
/**
 * @function getMaxFieldLength Gibt die max. Länge zurück und einen Fehlertext für die Formulardaten-Validation
 * @author Sebastian Weidner
 * @since 13.06.2024
 * @version 1.0
 *
 * @param maxLength Maximale Textlänge des input-Feldes
 */
const getMaxFieldLength = (maxLength: number): [number, string] => {
    const maxFieldLengthText = `Eingabefeld ist auf ${maxLength} Zeichen begrenzt`;
    return [maxLength, maxFieldLengthText];
};

const SignupSchema2 = yup.object({
    anrede: yup.string().required("Bitte geben Sie eine Anrede an!"),
    vorname: yup.string().trim().required("Bitte geben Sie Ihren Vornamen an!").max(...getMaxFieldLength(30)),
    nachname: yup.string().trim().required("Bitte geben Sie Ihren Nachnamen an!").max(...getMaxFieldLength(25)),
    land: yup.string().trim().required("Bitte geben Sie Ihr Herkunftsland an!"),
    bundesland: yup.string().trim().required("Bitte geben Sie Ihr Bundesland an!"),
    telefonnr: yup.string().trim().required("Bitte Geben Sie Ihre Telefonnummer an!").max(...getMaxFieldLength(25)),
    sprache: yup.array().min(1, "Bitte wählen Sie Ihre gesprochenen Sprachen aus!").required( "Bitte wählen Sie Ihre gesprochenen Sprachen aus!"),
    ueberMich: yup.string().trim().defined().max(...getMaxFieldLength(500))
});


const languages: string[] = [
    'Deutsch',
    'Englisch',
    'Französisch',
    'Italienisch',
    'Spanisch',
];


/**
 * @function SignupFormInputs2 Formular-Input-Felder für die Persönlichen Daten
 * @author Sebastian Weidner
 * @since 14.06.2024
 * @version 1.0
 *
 * @param ISignupForm2 "react-hook-form" Daten
 */
const SignupFormInputs2: React.FC<ISignupForm2> = ({form2}) => {

    const countries = ['Deutschland', 'Österreich', 'Schweiz'];
    const bundesland = ['Bayern', 'Hessen', 'Sachsen'];
    const sex = ['Herr', 'Frau', 'Divers'];


    //___________________ HTML: Formular ________________
    return (
        <>
            {/*--- Überschrift ---*/}
            <div className="border-b-2 border-gray-900/10 pb-4">
                <h3 className='text-xl font-bold mt-6 mb-0 text-gray-900 '>Persönliche Daten</h3>
                <p className="mt-1 pt-0 text-sm leading-6 text-gray-500">
                    Diese Informationen sind öffentlich sichtbar. Seien Sie vorsichtig, was Sie teilen.
                </p>
            </div>
            <div className="mt-6 mb-2 grid grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6">
                <InputSelect id="anrede" label="Anrede" fieldWidth={4} selectOptions={sex}
                             autoComplete="sex"
                             register={form2.register("anrede")} error={form2.formState.errors.anrede?.message}/>
                <InputField id="vorname" label="Vorname" type="text" autoComplete="given-name"
                            fieldWidth={4}
                            register={form2.register("vorname")} error={form2.formState.errors.vorname?.message}/>
                <InputField id="nachname" label="Nachname" type="text" autoComplete="family-name"
                            fieldWidth={4}
                            register={form2.register("nachname")} error={form2.formState.errors.nachname?.message}/>


                <InputSelect id="land" label="Land" fieldWidth={4} selectOptions={countries}
                             autoComplete="country-name"
                             register={form2.register("land")} error={form2.formState.errors.land?.message}/>
                <InputSelect id="bundesland" label="Bundesland" fieldWidth={4} selectOptions={bundesland}
                             register={form2.register("bundesland")} error={form2.formState.errors.bundesland?.message}/>

                <InputField id="telefonnr" label="Telefonnummer" type="tel" autoComplete="tel"
                            fieldWidth={4}
                            register={form2.register("telefonnr")} error={form2.formState.errors.telefonnr?.message}/>

                <InputMultiSelectDropdown label="Welche Sprachen sprechen Sie?" backgroundText="Sprachen auswählen..." selectOptions={languages} fieldWidth={4}
                                          control={form2.control} name="sprache" error={form2.formState.errors.sprache?.message}/>

                <InputTextarea id="ueberMich" label="Über mich" defaultValue=""
                               descr="Schreibe ein paar Sätze über dich." textboxRows={5}
                               register={form2.register("ueberMich")} error={form2.formState.errors.ueberMich?.message}/>

            </div>
        </>
    );
};



//__________ Exporte: ___________
export {
    SignupFormInputs2,
    SignupSchema2
}

export type {
    IFormInputs2
}
