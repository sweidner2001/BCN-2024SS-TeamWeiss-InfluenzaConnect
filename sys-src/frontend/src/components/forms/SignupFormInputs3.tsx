// React Imports:
import React, {useState} from 'react';
import * as yup from "yup";

// Imports eigene Componenten:
import InputFieldWithFixedText from "../input/InputFieldWithFixedText";




//___________________ Datentypen  ________________
interface ISignupForm1 {
    form3: any;
}

interface IFormInputs3 {
    instaUsername: string;
}




//___________________ Formular Validation  ________________
/**
 * @function getMaxFieldLength Gibt die max. Länge zurück und einen Fehlertext für die Formulardaten-Validation
 * @author Sebastian Weidner
 * @since 13.06.2024
 * @version 1.0
 *
 * @param maxLength Maximale Textlänge des input-Feldes
 */
const getMaxFieldLength = (maxLength: number): [number, string] => {
    const maxFieldLengthText = `Eingabefeld ist auf ${maxLength} Zeichen begrenzt`;
    return [maxLength, maxFieldLengthText];
};


const SignupSchema3 = yup.object({
    instaUsername: yup.string().trim().required("Bitte Geben Sie Ihren Instagram-Usernamen an!").max(...getMaxFieldLength(25))
});






/**
 * @function SignupFormInputs3 Formular-Input-Felder für die Verknüpfung von Socail-Media Accounts
 * @author Sebastian Weidner
 * @since 14.06.2024
 * @version 1.0
 *
 * @param ISignupForm3 "react-hook-form" Daten
 */
const SignupFormInputs3: React.FC<ISignupForm1> = ({form3}) => {

    //___________________ HTML: Formular ________________
    return (
        <>
            {/*--- Überschrift ---*/}
            <div className="border-b-2 border-gray-900/10 pb-4">
                <h3 className='text-xl font-bold mt-6 mb-0 text-gray-900 '>Verknüpfung Socail-Media Accounts</h3>
                <p className="mt-1 pt-0 text-sm leading-6 text-gray-500">
                    Diese Informationen dienen dazu, Sie als Influencer zu vermarkten und sind öffentlich sichtbar.
                </p>
            </div>
            <div className="mt-6 mb-2 grid grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6">

                <InputFieldWithFixedText register={form3.register("instaUsername")}
                                         error={form3.formState.errors.instaUsername?.message} fixedText="instagram.com/"
                                         id="instaUsername" label="Instagram Username" type="text"
                                         fieldWidth={4}/>
            </div>
        </>
    );
};



//__________ Exporte: ___________
export {
    SignupFormInputs3,
    SignupSchema3
}

export type {
    IFormInputs3
}
