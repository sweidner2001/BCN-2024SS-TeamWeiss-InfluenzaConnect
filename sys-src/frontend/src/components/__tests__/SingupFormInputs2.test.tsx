import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import { useForm, UseFormReturn } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { SignupFormInputs2, SignupSchema2, IFormInputs2 } from '../forms/SignupFormInputs2';
import userEvent from '@testing-library/user-event';
import {act} from 'react';

// Cleanup nach jedem Test
afterEach(cleanup);

const renderWithForm = (Component: React.FC<{ form2: UseFormReturn<any> }>) => {
    const Wrapper: React.FC = () => {
        const form2 = useForm<IFormInputs2>({
            resolver: yupResolver(SignupSchema2),
        });

        return <Component form2={form2} />;
    };

    return render(<Wrapper />);
};

test('renderstSignupFormInputs2 Komponente richtig', () => {
    renderWithForm(SignupFormInputs2);

    // Überprüfen der statischen Texte
    expect(screen.getByText('Persönliche Daten')).toBeInTheDocument();
    expect(screen.getByText('Diese Informationen sind öffentlich sichtbar. Seien Sie vorsichtig, was Sie teilen.')).toBeInTheDocument();

    // Überprüfen der Beschriftungen für die Eingabefelder
    expect(screen.getByLabelText('Anrede')).toBeInTheDocument();
    expect(screen.getByLabelText('Vorname')).toBeInTheDocument();
    expect(screen.getByLabelText('Nachname')).toBeInTheDocument();
    expect(screen.getByLabelText('In welchem Land leben Sie?')).toBeInTheDocument();
    expect(screen.getByLabelText('Telefonnummer')).toBeInTheDocument();

});



test('geht richtiig mit Input um', async () => {
    renderWithForm(SignupFormInputs2);

    const vornameInput = screen.getByLabelText('Vorname') as HTMLInputElement;
    fireEvent.change(vornameInput, { target: { value: 'Max' } });
    expect(vornameInput.value).toBe('Max');

    const nachnameInput = screen.getByLabelText('Nachname') as HTMLInputElement;
    fireEvent.change(nachnameInput, { target: { value: 'Mustermann' } });
    expect(nachnameInput.value).toBe('Mustermann');

    const telefonnrInput = screen.getByLabelText('Telefonnummer') as HTMLInputElement;
    fireEvent.change(telefonnrInput, { target: { value: '123456789' } });
    expect(telefonnrInput.value).toBe('123456789');
});

test('stimmt mit snapshot überein', () => {
    const { asFragment } = renderWithForm(SignupFormInputs2);
    expect(asFragment()).toMatchSnapshot();
});
