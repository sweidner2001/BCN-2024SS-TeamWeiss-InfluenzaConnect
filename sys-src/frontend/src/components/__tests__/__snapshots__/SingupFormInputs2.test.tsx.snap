// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stimmt mit snapshot überein 1`] = `
<DocumentFragment>
  <div
    class="border-b-2 border-gray-900/10 pb-4"
  >
    <h3
      class="text-xl font-bold mt-6 mb-0 text-gray-900 "
    >
      Persönliche Daten
    </h3>
    <p
      class="mt-1 pt-0 text-sm leading-6 text-gray-500"
    >
      Diese Informationen sind öffentlich sichtbar. Seien Sie vorsichtig, was Sie teilen.
    </p>
  </div>
  <div
    class="mt-6 mb-2 grid grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-6"
  >
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="anrede"
      >
        Anrede
      </label>
      <div
        class="mt-2"
      >
        <select
          autocomplete="sex"
          class="px-2 py-2 rounded-lg block w-full
 text-gray-900 placeholder:text-gray-400
 border-0 shadow-sm
 ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
          id="anrede"
          name="anrede"
        >
          <option
            value="Herr"
          >
            Herr
          </option>
          <option
            value="Frau"
          >
            Frau
          </option>
          <option
            value="Divers"
          >
            Divers
          </option>
        </select>
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="vorname"
      >
        Vorname
      </label>
      <div
        class="mt-2"
      >
        <input
          autocomplete="given-name"
          class="px-4 py-2 rounded-lg block w-full
 text-gray-900 placeholder:text-gray-400
 border-0 shadow-sm ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
          id="vorname"
          name="vorname"
          placeholder="Vorname"
          type="text"
        />
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="nachname"
      >
        Nachname
      </label>
      <div
        class="mt-2"
      >
        <input
          autocomplete="family-name"
          class="px-4 py-2 rounded-lg block w-full
 text-gray-900 placeholder:text-gray-400
 border-0 shadow-sm ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
          id="nachname"
          name="nachname"
          placeholder="Nachname"
          type="text"
        />
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="land"
      >
        In welchem Land leben Sie?
      </label>
      <div
        class="mt-2"
      >
        <select
          autocomplete="country-name"
          class="px-2 py-2 rounded-lg block w-full
 text-gray-900 placeholder:text-gray-400
 border-0 shadow-sm
 ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
          id="land"
          name="land"
        >
          <option
            value="Deutschland"
          >
            Deutschland
          </option>
          <option
            value="Österreich"
          >
            Österreich
          </option>
          <option
            value="England"
          >
            England
          </option>
          <option
            value="USA"
          >
            USA
          </option>
          <option
            value="Frankreich"
          >
            Frankreich
          </option>
          <option
            value="Italien"
          >
            Italien
          </option>
          <option
            value="Spanien"
          >
            Spanien
          </option>
        </select>
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="telefonnr"
      >
        Telefonnummer
      </label>
      <div
        class="mt-2"
      >
        <input
          autocomplete="tel"
          class="px-4 py-2 rounded-lg block w-full
 text-gray-900 placeholder:text-gray-400
 border-0 shadow-sm ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
          id="telefonnr"
          name="telefonnr"
          placeholder="Telefonnummer"
          type="tel"
        />
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="sm:col-span-4"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
      >
        Welche Sprachen sprechen Sie?
      </label>
      <div
        class="mt-2 relative"
      >
        <div
          class="px-4 py-2 rounded-lg block w-full cursor-pointer 
                                  border-0 shadow-sm ring-1 ring-inset outline-none ring-gray-300"
        >
          <span
            class="text-gray-400"
          >
            Sprachen auswählen...
          </span>
        </div>
        <div
          class="px-2 pt-1 h-4"
        />
      </div>
    </div>
    <div
      class="col-span-full"
    >
      <label
        class="block text-sm font-medium leading-6 text-gray-900"
        for="ueberMich"
      >
        Über mich
      </label>
      <p
        class="mt-1 text-sm leading-6 text-gray-600"
      >
        Schreibe ein paar Sätze über dich.
      </p>
      <textarea
        class="px-2 py-2 block w-full
 text-gray-900 text-sm placeholder:text-gray-400
 border-0 shadow-sm rounded-lg
 ring-1 ring-inset ring-gray-300
 focus:outline-none focus:ring-1 focus:ring-indigo-700"
        id="ueberMich"
        name="ueberMich"
        rows="5"
      />
      <div
        class="px-2 pt-1 h-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
